feat: new Kubernetes image registry patch (#149)
d2iq-labs/capi-runtime-extensions: docs/content/image-repository.md

---
title: "Image Repository"
---

Override the container image repository used when pulling Kubernetes images.

> To enable the API server certificate SANs enable the `extraapiservercertsansvars` and `extraapiservercertsanspatch`
> external patches on `ClusterClass`.

Reviewer: ```suggestion
```

```yaml
apiVersion: cluster.x-k8s.io/v1beta1
kind: ClusterClass
metadata:
  name: <NAME>
spec:
  patches:

>     - name: apiserver-cert-sans

Reviewer: ```suggestion
    - name: image-repository
```

Author: You're seeing how the sausage is made by most copy and pasting for new handlers, we would really benefit from a kubebuilder style generator.

      external:
        generateExtension: "imagerepositorypatch.capi-runtime-extensions"
        discoverVariablesExtension: "imagerepositoryvars.capi-runtime-extensions"
```

> On the cluster resource then specify desired certificate SANs values:

Reviewer: ```suggestion
On the cluster resource then specify desired Kubernetes image repository value:
```
